gaidds/React-crm-project
# I1BCIP 28 forgot your password

# Overview

This branch implements the frontend interface for the password reset functionality. It includes the UI for users to request a password reset and set a new password.

*NOTE*
To test this branch you need to run this branch in the backend: [I1BCIP-28-create-forgot-password-option](https://github.com/gaidds/Django-crm-project/pull/28).

---

## Endpoints

### Password Reset Request Form

- Path: /forgot-password
- Description: A form where users enter their email to request a password reset. This form interacts with the backend API to send the reset email.

### Password Reset Form

- Path: /reset-password/:uidb64/:token
- Description: A form where users can enter their new password using the token sent to their email. This form interacts with the backend API to reset the password.
---
## How to Test the Frontend

- Navigate to /login and click forgot password link to be redirected to /forgot-password.
- Enter a valid email address and submit the form.
- Check your email (ensure the backend is running) and click on the reset link.
- The reset link should open the /reset-forgot-password/:uidb64/:token page.
- Enter and confirm the new password and submit.
- Verify that the frontend handles success and error messages correctly.

## Expected Test Results

- Submitting an email should trigger the backend to send a password reset email.
- The reset password form should validate input and display appropriate messages for success and errors.
- After resetting the password, the user should be redirected to the login page with a success message.
